core/arch/arm/kernel/secure_partition.c

```diff
@@ -525,6 +526,92 @@ static TEE_Result handle_fdt_dev_regions(struct sp_ctx *ctx, void *fdt)
 	return TEE_SUCCESS;
 }
 
+#ifdef CFG_CORE_TPM_EVENT_LOG
+static TEE_Result handle_tpm_event_log(void *fdt, struct sp_session *s)
+{
+	int node = 0;
+	TEE_Result res = TEE_SUCCESS;
+	int prop_len = 0;
+	vaddr_t event_log_addr = 0;
+	size_t event_log_size = 0;
+	size_t page_count = 0;
+	struct fobj *f = NULL;
+	struct mobj *m = NULL;
+	struct sp_ctx *ctx = NULL;
+	uint32_t perm = TEE_MATTR_URW | TEE_MATTR_PRW;
+
+	node = fdt_node_offset_by_compatible(fdt, 0, "arm,tpm_event_log");
+	if (node < 0)
+		return TEE_SUCCESS;
+
+	/* Checking the existence and size of the event log properties */
+	if (!fdt_getprop(fdt, node, "tpm_event_log_addr", &prop_len) ||
+	    prop_len != sizeof(uint64_t)) {
+		EMSG("tpm_event_log_addr not found or has invalid size");
+		res = TEE_ERROR_BAD_FORMAT;
+		goto err;
+	}
+
+	if (!fdt_getprop(fdt, node, "tpm_event_log_size", &prop_len) ||
+	    prop_len != sizeof(uint32_t)) {
+		EMSG("tpm_event_log_size not found or has invalid size");
+		res = TEE_ERROR_BAD_FORMAT;
+		goto err;
```

Comment:
Nothing special is happening at `err:` so you could just return the error directly instead. This comment applies to all the `goro err;` below too.

Reply:
Done